docs/differences_from_aws.md

```diff
@@ -0,0 +1,101 @@
+# Differences from AWS Architecture
+
+This document explains how GCP Ingestion differs from the AWS implementation.
+
+[AWS Data Platform Architecture](https://mana.mozilla.org/wiki/display/SVCOPS/Telemetry+-+Data+Pipeline+Architecture)
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Replace Heka Message Format with PubSub Message Format](#replace-heka-message-format-with-pubsub-message-format)
+- [Replace EC2 Edge with Kubernetes Edge](#replace-ec2-edge-with-kubernetes-edge)
+- [Replace Kafka with PubSub](#replace-kafka-with-pubsub)
+- [Replace Hindsight Data Warehouse Loaders with Dataflow](#replace-hindsight-data-warehouse-loaders-with-dataflow)
+- [Replace S3 with Cloud Storage](#replace-s3-with-cloud-storage)
+- [Messages Always Delivered to Message Queue](#messages-always-delivered-to-message-queue)
+- [Landfill is Downstream from Message Queue](#landfill-is-downstream-from-message-queue)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+## Replace Heka Message Format with PubSub Message Format
+
+Heka messages are protobuf, and have a special framing format. PubSub messages
+are newline delimited JSON. The PubSub message format is being used both because
```

Comment:
"PubSub messsages are newline delimited JSON" doesn't sound right to me. The GCS files we store will be in ndjson format, but a Pubsub message itself does not necessarily contain JSON, and even when it does, it's probably a single json message rather than ndjson.

Perhaps this secion should be framed as "Replace Heka Message Format with Compressed Pubsub-formatted JSON"

Reply:
part of the problem is that the heka message format and the pubsub message format aren't actually what's being swapped out here. it's heka framed protobuf vs compressed ndjson, and the json schema varies. i'll update this section to reflect that.